bedita-app/vendors/shells/tasks/cleanup_data.php

```diff
@@ -0,0 +1,134 @@
+ * If not, see <http://gnu.org/licenses/lgpl-3.0.html>.
+ *
+ *------------------------------------------------------------------->8-----
+ */
+
+/**
+ * Cleanup data task
+ * 
+ * @property \BEobject $BEobject
+ */
+class CleanupDataTask extends BeditaBaseShell {
+    
+    /**
+     * @inheritDoc
+     */
+    public $uses = ['BEObject'];
+
+    protected $truncateTables = ['event_logs', 'mail_jobs', 'mail_logs'];
+
+    protected $cleanupTables = ['history', 'versions'];
+
+    public function help() {
+        $this->hr();
+        $this->out('cleanup data script shell usage:');
+        $this->out('');
+        $this->out('./cake.sh cleanupData // truncate event_logs, mail_jobs, mail_logs, clean old data from history and versions');
+        $this->out('./cake.sh cleanupData -ld <limit date yyyy-MM-dd> // use custom limit date for clean');
+        $this->out('./cake.sh cleanupData -ni // no interactive mode');
+        $this->out('./cake.sh cleanupData -tt <comma separated table names> // truncate event_logs, mail_jobs, mail_logs + more tables');
+        $this->out('./cake.sh cleanupData help // show this help');
+        $this->out('');
+    }
+
+    /**
+     * Tables to clean (truncate or delete data):
+     * - event_logs (truncate)
+     * - mail_jobs (truncate, if no messages pending)
+     * - mail_logs (truncate)
+     * - history (cleanup: mantain only "CY-1 / CY" data (CY is the Current Year))
+     * - versions (cleanup: mantain only "CY-1 / CY" data (CY is the Current Year))
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        $this->hr();
+        $this->out('BEdita Data Cleanup');
+        $this->out('--- count data before cleaning ---');
+        $tables = $this->truncateTables;
+        if (isset($this->params['tt'])) {
+            $tables = array_merge($tables, explode(',', $this->params['tt']));
+        }
+        $countTables = array_merge($this->cleanupTables, $tables);
+        sort($countTables);
+        $limitDate = date('Y')-2 . '-12-31 23:59:59';
+		if (isset($this->params['ld'])) {
+			$limitDate = $this->params['ld'] . ' 23:59:59';
+        }
+        $limitWhere = sprintf('WHERE created < \'%s\'', $limitDate);
+        foreach ($countTables as $tableName) {
+            $this->countRecords($tableName);
+            $this->countRecords($tableName, $limitWhere);
+        }
+        // check if mail_jobs can be truncated
+        $query = 'SELECT COUNT(*) AS n FROM mail_jobs WHERE status NOT IN (\'sent\', \'failed\')';
+        $result = $this->BEObject->query($query);
+        $pending = $result[0][0]['n'];
+        $this->out(sprintf('%s => %s', $query, $pending));
+        if ($pending > 0) {
+            $this->out('mail_jobs won\'t be truncated, because there are pending jobs. cleanup instead');
+            if (($key = array_search('mail_jobs', $tables)) !== false) {
+                unset($tables[$key]);
+            }
+            $this->cleanupTables[] = 'mail_jobs';
```

Comment:
Adding this table to the ones for cleanup will allow deletion of jobs in a pending status

Reply:
I assume that there aren't pending jobs from 1 year or so (cleanup works with a limit date, default 31.12.`current year - 2`